repos/spack_repo/builtin/packages/dftd4/package.py

```diff
@@ -56,6 +56,8 @@ class Dftd4(MesonPackage, CMakePackage):
         depends_on(f"mctc-lib build_system={build_system}", when=f"build_system={build_system}")
         depends_on(f"multicharge build_system={build_system}", when=f"build_system={build_system}")
 
+    depends_on("mctc-lib@0.3", when="@:3.8")
+    depends_on("multicharge@0.3", when="@:3.8")
```

Comment:
It depends only on these specific versions? Or it should be a version range?

Reply:
The versions of the dependencies are locked unfortunately. The developers also change the api of these libraries which lead to compilation failure.